packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -409,6 +410,23 @@ export class AuthUtil {
 
         return state
     }
+
+    /**
+     * Edge Case: Due to a change in behaviour/functionality, there are potential extra
+     * auth connections that the Amazon Q extension has cached. We need to remove these
+     * as they are irrelevant to the Q extension and can cause issues.
+     */
+    public async clearExtraConnections(): Promise<void> {
```

Comment:
Is this concept actually specific to Q/CW or could it live in the top-level auth.ts module (or similar)?

Although it's only called by Q currently, if it's needed for non-Q in the future, it's more discoverable if it lives in the base auth.ts module.